File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    var answer = '';
    const pMap = {};
    
    for (const p of participant) {
        pMap[p] = (pMap[p] || 0) + 1;
    }
    
    for (const c of completion) {
        pMap[c] -= 1;
    }
    
    for (const p in pMap) {
        if (pMap[p] > 0) answer = p;
    }
    
    return answer;
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    answer = ''
    dict = {}
    for p in participant:
        if p in dict:
            dict[p] += 1
        else:
            dict[p] = 1
    for c in completion:
        if c in dict:
            dict[c] -= 1
    for p in dict:
        if dict[p] == 1:
            answer = p
    return answer